--- a/modules/bar/components/workspaces/Workspace.qml
+++ b/modules/bar/components/workspaces/Workspace.qml
@@ -112,6 +112,11 @@
         StyledText {
             animate: true
             text: {
+                const icons = {"1":"󰖟","2":"","3":"3","4":"4","5":"","6":"󰮂","7":"󰈮","8":"󰅨","9":"󰅱","10":"10"};
+                const wsIcon = icons[root.ws.toString()];
+                if (wsIcon && root.isOccupied)
+                    return wsIcon;
+
                 if (root.focused) {
                     const label = root.activeLabel;
                     if (label)
@@ -141,6 +146,7 @@
             color: root.fgColour
             verticalAlignment: Qt.AlignVCenter
             font.family: Tokens.font.workspaces
+            font.pointSize: Tokens.font.icon.small.pointSize
         }
     }
 
